bob-common/src/mapper.rs

```diff
@@ -134,15 +134,14 @@ impl Virtual {
         let mut support_nodes = Vec::with_capacity(count);
         for (id, node) in self.nodes.iter().skip(offset % self.nodes.len()) {
             if target_nodes.iter().all(|i| i.index() != *id) {
-                support_nodes.push(node);
-                if support_nodes.len() >= count {
-                    break;
+                if support_nodes.len() < count && node.connection_available() {
```

Comment:
This is not good:
1. Iterations continue even when `support_nodes` are full. Previously there was a `break` to stop the iteration
2. This ignores the idea of uniform alien distribution, that was introduced by https://github.com/qoollo/bob/pull/629. Imagine the situation when in 4 nodes cluster, nodes 1 and 2 are tuned off. According to the implemented logic node 3 will keep 3 times more aliens than node 4. We need to make a uniform distribution, taking into account the availability of the connection